### ISSUE TITLE
More detailed description of the data construction process

### ISSUE THREAD

**Issue by Yimi81** [MeetKai/functionary#89]
@musabgultekin Thank you for your excellent work. I would like to refer to the rough process you provided to create my own parallel function call fine-tuning dataset, but the description [here ](https://github.com/MeetKai/functionary/issues/57#issuecomment-1826734192) is quite simple. Can you provide a more detailed process and some samples of the final dataset.

**Comment by musabgultekin** [comment]
Hi,
Are you from the core developers of Yi? Cause we were also considering Fine-Tuning on Yi34B. 
In regards to the dataset, it heavily depends on the goal. We wanted to generate something generic for all use cases. Are you considering something generic or some specific use-case/industry? Cause the steps might differ, depending on the goal.

**Comment by Yimi81** [comment]
Thank you for your reply. I am Yi's developer, but not the core. I am responsible for supporting the open source community. The community has always hoped that the official can improve the function call capability of the Yi model. So I wanted to fine tune a Yi-34B Chat model that supports parallel tool calls, and then I found this repository and tested the model's performance, which met my expectations very well. generic is what I am considering

**Comment by Yimi81** [comment]
If you consider fine-tuning Yi-34B, that would be great. The Yi community needs such a model

**Comment by Yimi81** [comment]
@musabgultekin

**Comment by khai-meetkai** [comment]
@Yimi81 
We already finetuned a model based on Yi-34B with the same data we trained for small and medium models. However, from evaluation, the performance of the model based on Yi-34B is **just slightly better than** as our small model (based on Mistral-7B), so didn't get any substantial benefits. Therefore, we didn't release the model.

**Comment by Yimi81** [comment]
@khai-meetkai hi khai, omorrow we will launch the Yi-1.5 series model, which has achieved excellent results in math and code. May I ask if you can try our v1.5 model again for fine-tuning? The Yi community needs a model that supports function call.
![1715515056858](https://github.com/MeetKai/functionary/assets/66633207/f2f7bc85-7352-4d09-9f1e-e992490d6c17)

**Comment by musabgultekin** [comment]
The results look really nice. We'll definitely consider especially if the context size is more than 8k.

**Comment by musabgultekin** [comment]
Could you ping us when its released and we'll check it out

**Comment by musabgultekin** [comment]
Also, could you provide a recommended LR for tuning? Cause we found out that especially llama 3 performance is highly sensitive to the peak LR (we used rate scheduler with warmup)

**Comment by khai-meetkai** [comment]
Hi @Yimi81 , please let us know when your model is released !

**Comment by Yimi81** [comment]
@musabgultekin @khai-meetkai  our models have been released, and the current context is 4k. (we are discussing releasing a model of size >=16k internally.) Additionally, I apologize that I am not clear about the peak LR information, and I will inquire with my colleagues.

**Comment by Yimi81** [comment]
Models with longer context have been released：
✅ Yi-1.5-34B-Chat-16k
✅ Yi-1.5-34B-Base-32k
✅ Yi-1.5-9B-Chat-16k
✅ Yi-1.5-9B-Base-32k
https://huggingface.co/collections/01-ai/yi-2023-11-663f3f19119ff712e176720f
